From b3a63df49d19cd4d98f34a5d360515976fbe1224 Mon Sep 17 00:00:00 2001
From: Udipto Goswami <udipto.goswami@oss.qualcomm.com>
Date: Thu, 19 Mar 2026 17:33:43 +0530
Subject: [PATCH 40/63] ANDROID: abi_gki_aarch64_qcom: whitelist some xhci &
 traceiter symbols

Whitelist the below symbols:
__traceiter_android_vh_xhci_handle_offload
xhci_handle_event_trb

These are required for offload feature support enablingto use the same
event handling framework in vendor offload solutions.

Bug: 494307525
Bug: 487900539
Bug: 497692253
Change-Id: Ib9bae4dae908d9a671cd799d90506df474b0b225
Signed-off-by: Udipto Goswami <udipto.goswami@oss.qualcomm.com>
[jstultz: Add missing __tracepoint_android_vh_xhci_handle_offload
symbol and abi.stg file]
Signed-off-by: John Stultz <jstultz@google.com>
---
 gki/aarch64/abi.stg      | 45 ++++++++++++++++++++++++++++++++++++++++
 gki/aarch64/symbols/qcom |  3 +++
 2 files changed, 48 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index d9f2bee52ed9..7ef34e7607c8 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -411847,6 +411847,13 @@ function {
   parameter_id: 0xc9082b19
   parameter_id: 0x5a8462fa
 }
+function {
+  id: 0x9917f312
+  return_type_id: 0x6720d32f
+  parameter_id: 0x1285100d
+  parameter_id: 0x0e88ae5c
+  parameter_id: 0x31da1e83
+}
 function {
   id: 0x991a5468
   return_type_id: 0x6720d32f
@@ -420759,6 +420766,14 @@ function {
   parameter_id: 0x11b57133
   parameter_id: 0x33756485
 }
+function {
+  id: 0x9be64b37
+  return_type_id: 0x6720d32f
+  parameter_id: 0x18bd6530
+  parameter_id: 0x1285100d
+  parameter_id: 0x0e88ae5c
+  parameter_id: 0x11cfee5a
+}
 function {
   id: 0x9be64bdf
   return_type_id: 0x6720d32f
@@ -446018,6 +446033,15 @@ elf_symbol {
   type_id: 0x9a3fe476
   full_name: "__traceiter_android_vh_wq_lockup_pool"
 }
+elf_symbol {
+  id: 0xbc1f9cf9
+  name: "__traceiter_android_vh_xhci_handle_offload"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xa0a4cf48
+  type_id: 0x9be64b37
+  full_name: "__traceiter_android_vh_xhci_handle_offload"
+}
 elf_symbol {
   id: 0x18619e65
   name: "__traceiter_android_vh_zs_shrinker_adjust"
@@ -453848,6 +453872,15 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_vh_wq_lockup_pool"
 }
+elf_symbol {
+  id: 0x87ed67f3
+  name: "__tracepoint_android_vh_xhci_handle_offload"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_xhci_handle_offload"
+}
 elf_symbol {
   id: 0x85ad7f9f
   name: "__tracepoint_android_vh_zs_shrinker_adjust"
@@ -525182,6 +525215,15 @@ elf_symbol {
   type_id: 0xcbd33190
   full_name: "xhci_get_ep_ctx"
 }
+elf_symbol {
+  id: 0x1bc8d75d
+  name: "xhci_handle_event_trb"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x5edaf9e1
+  type_id: 0x9917f312
+  full_name: "xhci_handle_event_trb"
+}
 elf_symbol {
   id: 0x71f2808e
   name: "xhci_init_driver"
@@ -527216,6 +527258,7 @@ interface {
   symbol_id: 0x892f6a60
   symbol_id: 0x20e7f214
   symbol_id: 0xae5e5469
+  symbol_id: 0xbc1f9cf9
   symbol_id: 0x18619e65
   symbol_id: 0xc4d6f652
   symbol_id: 0x6911084f
@@ -528086,6 +528129,7 @@ interface {
   symbol_id: 0x449c21a2
   symbol_id: 0xf1bbc512
   symbol_id: 0xa13f65ff
+  symbol_id: 0x87ed67f3
   symbol_id: 0x85ad7f9f
   symbol_id: 0x93f06708
   symbol_id: 0xf57e8f65
@@ -536007,6 +536051,7 @@ interface {
   symbol_id: 0xc46f1f1f
   symbol_id: 0xf30af7b3
   symbol_id: 0xc9a9e8ab
+  symbol_id: 0x1bc8d75d
   symbol_id: 0x71f2808e
   symbol_id: 0x246bf5fe
   symbol_id: 0x4ae6dff8
diff --git a/gki/aarch64/symbols/qcom b/gki/aarch64/symbols/qcom
index 8984f4005629..703090fd8ec4 100644
--- a/gki/aarch64/symbols/qcom
+++ b/gki/aarch64/symbols/qcom
@@ -3480,6 +3480,7 @@
   __traceiter_android_vh_ufs_send_uic_command
   __traceiter_android_vh_ufs_update_sdev
   __traceiter_android_vh_update_topology_flags_workfn
+  __traceiter_android_vh_xhci_handle_offload
   __traceiter_binder_transaction_received
   __traceiter_cpu_frequency_limits
   __traceiter_cpu_idle
@@ -3628,6 +3629,7 @@
   __tracepoint_android_vh_ufs_send_uic_command
   __tracepoint_android_vh_ufs_update_sdev
   __tracepoint_android_vh_update_topology_flags_workfn
+  __tracepoint_android_vh_xhci_handle_offload
   __tracepoint_binder_transaction_received
   __tracepoint_cpu_frequency_limits
   __tracepoint_cpu_idle
@@ -4117,6 +4119,7 @@
   xfrm_lookup
   xhci_create_secondary_interrupter
   xhci_get_endpoint_index
+  xhci_handle_event_trb
   xhci_initialize_ring_info
   xhci_remove_secondary_interrupter
   xhci_sideband_add_endpoint
-- 
2.50.1

